ext/tunemygc/tunemygc_ext.c

```diff
@@ -39,26 +43,53 @@ static VALUE tunemygc_walltime(VALUE mod)
     return DBL2NUM(_tunemygc_walltime());
 }
 
+static void reverse(tunemygc_stat_record** head_ref) {
```

Comment:
You could just have a static pointer to the first record, as well as the current record. I think this would allow you to get rid of `reverse`.